File: editor/en/[2962]Count Subarrays Where Max Element Appears at Least K Times.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        max_num = max(nums)
        count = 0
        left = 0
        freq = 0

        for right in range(len(nums)):
            if nums[right] == max_num:
                freq += 1

            while freq >= k:
                if nums[left] == max_num:
                    freq -= 1
                left += 1

            count += left

        return count
    # leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2962]Count Subarrays Where Max Element Appears at Least K Times.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int n = nums.length;
        int max = 0;

        for (int num : nums) {
            max = Math.max(max, num);
        }

        long count = 0;
        int left = 0;
        int freq = 0;

        for (int right = 0; right < n; right++) {
            if (nums[right] == max) {
                freq++;
            }

            while (freq >= k) {
                count += (n - right);

                if (nums[left++] == max) {
                    freq--;
                }
            }
        }

        return count;
    }
}


//leetcode submit region right(Prohibit modification and deletion)
